tools/topology/m4/pga.m4

```diff
@@ -6,13 +6,14 @@ dnl PGA name)
 
 define(`N_PGA', `PGA'PIPELINE_ID`.'$1)
 
-dnl W_PGA(name, format, periods_sink, periods_source, kcontrol0. kcontrol1...etc)
+dnl W_PGA(name, format, periods_sink, periods_source, core, kcontrol0. kcontrol1...etc)
```

Comment:
Btw, unrelated (and not for this PR) - can you start thinking about how we can reduce the number of arguments for each macro. I'm thinking given the W_PGA example above we could have
```
W_PERIODS(name, PGA, 2,2)
W_FORMAT(name, PGA, s32le)
W_KCONTROLS(name, PGA, ...)
```
This would then define macros 
```
define(`PGA_name_PERIODS_SOURCE`, `2`)
```
and so on.
These could all then be brought together and instantiated using
```
W_PGA(name)
```
Which then uses all the PGA_name_ macros to construct the PGA. This also means if we dont define a W_CORE() macro then the default value is used or it's left blank. This means it becomes much easier to expand the functionality and simplifies usage.

Reply:
@lyakh any thoughts on this ? We should probably make it a feature.

Reply:
@lgirdwood we have several levels of m4 files:
1. those, defining specific primitives in `common`, `m4`, and `platform`
2. those, defining pipelines in `sof`, using primitives from (1)
3. those, defining specific topologies in the root `topology` directory, using (2) and some of (1)

The "end user," anybody, defining or modifying a machine topology works with level (3), they hardly are expected to touch levels (1) or (2). Macros like `W_PGA()` are mostly used in level (2), they're hardly used directly in level (3). Do you propose to rework all levels or only those, used on level (3)? That would seem the most important simplification to me to make end topologies easier to work with? Also it would be good to be able to re-use definitions. In your proposal you have to define parameters for each `name`. What if we define several pipelines in a level (3) topology and some of them share the same values of parameters, like formats and sampling rate. It would be good to set such parameters once for all pipelines until they are redefined? E.g.
```
W_FORMAT(PIPELINE, s32le)
W_...
PIPELINE_PCM_ADD(1, 0)
PIPELINE_PCM_ADD(2, 1)
```
As for level (2) and (1) macros - we can simplify them too, but since they aren't exposed to end-users, maybe it's less important?

Reply:
Good idea, the three levels make a lot of sense and we should capture this in the docs. We should also make it compulsory that  new components provide levels 1 & 2 for that component.
Yes, good idea on the name where the pipelines are the same (i.e. HDMI). Lets do level 3 first.